### PR TITLE
fix:remove active components from the dropdown and sort with componentId

### LINKED ISSUE #5876: Bug: Component Type
Issue:
Selection of Component ID/Type is not associated with the selected system

Steps to Recreate:

- Navigate to the MP module
- Check out 564 config CCB
- Select Systems from Sections in MP workspace
- Select the view/edit for the following system: System ID: 100, System Type: NOX, System Designation: P, Fuel Code: NFS, Begin Date/Time: 2009-11-20
- Scroll down and click on View/Edit for the data with  component type  set to PRB
- Add End date 10/31/2023 hour 0
- Click Add Component

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNUNTQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1c13c7c421c007493fb1a7e59a0ecc2f99a6df17/image.png)

### PR DESCRIPTION
**Pull request by ergjustin**
## Ticket 
[Remove active component from dropdown](https://github.com/US-EPA-CAMD/easey-ui/issues/5876#issuecomment-2064989303)

## Change

1. Remove active component from dropdown
2. Sort dropdown list by componentId

### CODE REVIEW

File: src/components/ModalAddComponent/ModalAddComponent.js [hunk 1]
```diff
@@ -38,14 +38,20 @@ const ModalAddComponent = ({
...
       const sysWithEndDate = sysComps.filter((sy) => sy.endDate && new Date(sy.endDate) < new Date());
 
+      //Filtering system component with active | no endDate
+      const activeSystem = sysComps.filter((sy) => !sy.endDate || new Date(sy?.endDate) > new Date());
+
       // selecting component from main. that has componentId is equal to sysWithEndDate's componentId
       const componetWithSystemEndDate = main.filter(({ componentId }) => sysWithEndDate.some(({ componentId: sysCompId }) => sysCompId === componentId));
 
+      // remove active components 
+      const componentWithNonActive = componetWithSystemEndDate.filter(({ componentId, componentTypeCode }) => !activeSystem.some(({ componentId: sysCompId, componentTypeCode: sysCompType }) => sysCompId === componentId && componentTypeCode === sysCompType))
```
Comment (maxdiebold-erg) [inline src/components/ModalAddComponent/ModalAddComponent.js]: Is it necessary to compare the `componentTypeCode`, or is the `componentId` alone a unique identifier? if both are necessary for uniqueness, then should the comparison be added to line 45 above?

**Review by maxdiebold-erg — COMMENTED** [review]
Looks like this fixes the issue, let's hope this one sticks.

I have one question below for clarification.

File: src/components/ModalAddComponent/ModalAddComponent.js (hunk 1 above)
Comment (ergjustin) [reply src/components/ModalAddComponent/ModalAddComponent.js]: `componentTypeCode` is not necessary for uniqueness.
I will remove the `componentTypeCode` from the uniqueness and modify the sorting

**Review by maxdiebold-erg — APPROVED** [review]
Looks good!

**Pull request merged**